File: backend/auth2/src/main/java/com/ewatts/auth2/dto/JwtResponse.java
package com.ewatts.auth2.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JwtResponse {
    private String accessToken;
    private String refreshToken;
    private String tokenType = "Bearer";
    private Long expiresIn;
    private UserDto user;

    public JwtResponse(String accessToken, String refreshToken, Long expiresIn, UserDto user) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.expiresIn = expiresIn;
        this.user = user;
    }
}

File: my_app/lib/models/auth/jwt_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:my_app/models/auth/user_dto.dart';

part 'jwt_response.g.dart';

@JsonSerializable()
class JwtResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int expiresIn;
  final UserDto user;

  JwtResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    required this.user,
  });

  factory JwtResponse.fromJson(Map<String, dynamic> json) =>
      _$JwtResponseFromJson(json);
  Map<String, dynamic> toJson() => _$JwtResponseToJson(this);
}
